packages/central-server/__tests__/sync/CentralSyncManager.test.js

```diff
@@ -2838,4 +2838,965 @@ describe('CentralSyncManager', () => {
+            sensitiveFacility.id,
+            models.LabRequest.tableName,
+          );
+          expect(labRequestIds).toContain(sensitiveLabRequest.id);
+          expect(labRequestIds).toContain(nonSensitiveLabRequest.id);
+        });
+      });
+    });
+
+    describe('edge cases', () => {
+      it('wont sync between facilities just because they are both sensitive', async () => {
+        const sensitiveFacilityA = await models.Facility.create(
+          fake(models.Facility, {
+            isSensitive: true,
+          }),
+        );
+        const sensitiveFacilityB = await models.Facility.create(
+          fake(models.Facility, {
+            isSensitive: true,
+          }),
+        );
+        const sensitiveLocationA = await models.Location.create(
+          fake(models.Location, {
+            facilityId: sensitiveFacilityA.id,
+          }),
+        );
+        const sensitiveDepartmentA = await models.Department.create(
+          fake(models.Department, {
+            facilityId: sensitiveFacilityA.id,
+          }),
+        );
+        const sensitiveLocationB = await models.Location.create(
+          fake(models.Location, {
+            facilityId: sensitiveFacilityB.id,
+          }),
+        );
+        const sensitiveDepartmentB = await models.Department.create(
+          fake(models.Department, {
+            facilityId: sensitiveFacilityB.id,
+          }),
+        );
+        const sensitiveEncounterA = await models.Encounter.create({
+          ...fake(models.Encounter),
+          patientId: patient.id,
+          locationId: sensitiveLocationA.id,
+          departmentId: sensitiveDepartmentA.id,
+          examinerId: practitioner.id,
+          endDate: null,
+        });
+        const sensitiveEncounterB = await models.Encounter.create({
+          ...fake(models.Encounter),
+          patientId: patient.id,
+          locationId: sensitiveLocationB.id,
+          departmentId: sensitiveDepartmentB.id,
+          examinerId: practitioner.id,
+          endDate: null,
+        });
+
+        const centralSyncManager = initializeCentralSyncManager(lookupEnabledConfig);
+        await centralSyncManager.updateLookupTable();
+
+        const { sessionId } = await centralSyncManager.startSession();
+        await waitForSession(centralSyncManager, sessionId);
+
+        await centralSyncManager.setupSnapshotForPull(
+          sessionId,
+          {
+            since: 1,
+            facilityIds: [sensitiveFacilityA.id],
+          },
+          () => true,
+        );
+
+        const outgoingChanges = await centralSyncManager.getOutgoingChanges(sessionId, {});
+        const encounterIds = outgoingChanges
+          .filter(c => c.recordType === 'encounters')
+          .map(c => c.recordId);
+
+        expect(encounterIds).toContain(sensitiveEncounterA.id);
+        expect(encounterIds).not.toContain(sensitiveEncounterB.id);
+      });
```

Comment:
Very quick skim through these tests, didn't really read them after the first few as they were all so similar